### PR TITLE
Issue 780 Stage 1 - [ENCHANCEMENT] upgrade project to use BindView

### PR DESCRIPTION
**Pull request by RyanMarzec**
## Description

I have begun the work to upgrade the project to use ButterKnife @BindView instead of FindViewById.
Some sections that were not using Butterknife I updated to use a ViewHolder class for the variables.

Other classes that were already using BindView I change to follow the format being used that I will update at a later stage to use ViewHolder classes so that there is a consistent implementation and style.

Lastly, there are still some more findViewById that were either casted or embedded in a snackBar call that my initial grep to search for "= FindViewById" that I will look at in the next stage.

Fixes #(issue)

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 1]
```diff
@@ -87,21 +90,27 @@
 
     private SharedPreferences mSharedPreferences;
     private String mToken;
-    private DrawerLayout mDrawer;
+    //private DrawerLayout mDrawer;
     private Handler mHandler;
-    private NavigationView mNavigationView;
+    //private NavigationView mNavigationView;
```
Comment (prabhakar267) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: Do not comment these, if they are no longer used, remove them

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 2]
```diff
@@ -87,21 +90,27 @@
...
     private int mPreviousMenuItemId;
     private static final String travelShortcut = "io.github.project_travel_mate.TravelShortcut";
     private static final String myTripsShortcut = "io.github.project_travel_mate.MyTripsShortcut";
     private static final String utilitiesShortcut = "io.github.project_travel_mate.UtilitiesShortcut";
 
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+    @BindView(R.id.drawer_layout)
+    DrawerLayout mDrawer;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: private member variable names should start with `m` 
this isn't private.
`drawerlayout`

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java [hunk 3]
```diff
@@ -75,7 +75,14 @@
     TextView selectedItemAddress;
     @BindView(R.id.item_info)
     LinearLayout linearLayout;
+    @BindView(R.id.bottom_sheet)
+    LinearLayout layoutBottomSheet;
+    @BindView(R.id.map)
+    MapView mMap;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: `mapView`

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/funfacts/FunFactsActivity.java [hunk 4]
```diff
@@ -27,6 +27,7 @@
 import butterknife.BindView;
 import butterknife.ButterKnife;
 import io.github.project_travel_mate.R;
+import io.reactivex.annotations.Nullable;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/destinations/funfacts/FunFactsActivity.java]: remove this

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/funfacts/FunFactsActivity.java [hunk 5]
```diff
@@ -38,33 +39,32 @@
...
     private Handler mHandler;
     Boolean isFirstVisitEnd = true;
     public int scrollState;
 
-
+    @BindView(R.id.vp)
+    ViewPager viewPager;
+    @BindView(R.id.animation_view)
+    LottieAnimationView animationView;
+    @Nullable
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/destinations/funfacts/FunFactsActivity.java]: why this?

File: Android/app/src/main/java/io/github/project_travel_mate/travel/MapViewRealTimeActivity.java [hunk 6]
```diff
@@ -69,12 +69,17 @@
     LottieAnimationView animationView;
     @BindView(R.id.layout)
     LinearLayout layout;
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+    @BindView(R.id.map)
+    MapView mMap;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/MapViewRealTimeActivity.java]: `mapView`

File: Android/app/src/main/java/io/github/project_travel_mate/travel/MapViewRealTimeActivity.java [hunk 7]
```diff
@@ -69,12 +69,17 @@
...
+
     private int mIndex = 0;
     private Handler mHandler;
     private String mToken;
     private String mCurlat;
     private String mCurlon;
-    private MapView mMap;
+    //private MapView mMap;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/MapViewRealTimeActivity.java]: dont comment. remove

File: Android/app/src/main/java/io/github/project_travel_mate/travel/ShoppingCurrentCityActivity.java [hunk 8]
```diff
@@ -62,8 +62,11 @@
     LinearLayout layout;
     @BindView(R.id.show_city_name)
     TextView showCityName;
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+    @BindView(R.id.search_view)
+    MaterialSearchView mSearchView;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/ShoppingCurrentCityActivity.java]: `searchView`

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyListViewActivity.java [hunk 9]
```diff
@@ -16,13 +16,14 @@
...
 import butterknife.BindView;
+import butterknife.ButterKnife;
 import io.github.project_travel_mate.R;
 import objects.ZoneName;
 
 public class CurrencyListViewActivity extends Activity implements TextWatcher {
 
     @BindView(R.id.listView)
-    RecyclerView mListview;
+    RecyclerView mListView;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyListViewActivity.java]: as you are changing name here, change to `listView`

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java [hunk 10]
```diff
@@ -119,7 +118,6 @@ public boolean onCreateOptionsMenu(Menu menu) {
 
     public void createCode() {
 
-        ImageView qrCodeView = findViewById(R.id.im);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java]: is this not needed?

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/TimezoneListViewActivity.java [hunk 11]
```diff
@@ -24,7 +24,7 @@
 public class TimezoneListViewActivity extends Activity implements TextWatcher {
 
     @BindView(R.id.listView)
-    RecyclerView mListview;
+    RecyclerView mListView;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/TimezoneListViewActivity.java]: same here...

**Comment by RyanMarzec** [comment]
I have been busy with school assignments but ill fix these changes soon as I can!

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/funfacts/FunFactsActivity.java (hunk 5 above)
Comment (RyanMarzec) [reply Android/app/src/main/java/io/github/project_travel_mate/destinations/funfacts/FunFactsActivity.java]: `Caused by: java.lang.IllegalStateException: Required view 'toolbar' with ID 2131362437 for field 'toolbar' was not found. If this view is optional add '@Nullable' (fields) or '@Optional' (methods) annotation.`
App crashes when opening the FunFactsActivity without this being labeled nullable.

I took a look at the XML file and ran built the app in master, and there is actually no toolbar on this page. is this something that was broken in a previous pull request or always the case? if latter I can just remove toolbar altogether from this page.

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/funfacts/FunFactsActivity.java (hunk 4 above)
Comment (RyanMarzec) [reply Android/app/src/main/java/io/github/project_travel_mate/destinations/funfacts/FunFactsActivity.java]: Same as below.

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java (hunk 10 above)
Comment (RyanMarzec) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java]: no, don't believe so. It is declared at the top with BindView, I can do some more testing over the weekend and create a second account to test the QR codes before merging if you would like.

**Comment by RyanMarzec** [comment]
@Swati4star @prabhakar267 hopefully I didn't miss anything, there are just the issues in FunFactsActivity and ShareContactActivity I need your feedback on first.

File: Android/app/src/main/java/io/github/project_travel_mate/travel/ListViewRealTimeActivity.java [hunk 12]
```diff
@@ -30,6 +30,8 @@
     TabLayout tabLayout;
     @BindView(R.id.viewpager)
     ViewPager viewPager;
+    @BindView(R.id.action_map_view)
+    MapView mMap;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/ListViewRealTimeActivity.java]: change to `mapView`

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/ChecklistActivity.java [hunk 13]
```diff
@@ -51,4 +54,13 @@ public static Intent getStartIntent(Context context) {
         Intent intent = new Intent(context, ChecklistActivity.class);
         return intent;
     }
+
+    class ViewHolder {
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/ChecklistActivity.java]: This is an activity. Why are we using a viewholder here?

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java (hunk 10 above)
Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java]: just test if this works fine 👍

File: Android/app/src/main/java/utils/TravelMateAnalogClock.java [hunk 14]
```diff
@@ -215,8 +218,8 @@ public TravelMateAnalogClock setDiameterInPixels(int diameterInPixels) {
 
     /**
      * @param opacity: ranges from 0 (transparent) to 1.0 (opaque)
-     *
-     *               Default: 1.0f
+     *                 <p>
```
Comment (Swati4star) [inline Android/app/src/main/java/utils/TravelMateAnalogClock.java]: remove this line

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java (hunk 10 above)
Comment (RyanMarzec) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java]: travel_mate_test
I created a second account using an emulator and was able to scan the QR code using my mobile device, everything seems to be working.

**Comment by RyanMarzec** [comment]
```
$ ./gradlew assembleDebug assembleRelease
....
BUILD SUCCESSFUL in 6m 49s
55 actionable tasks: 37 executed, 18 up-to-date

$ ./gradlew checkstyle

BUILD SUCCESSFUL in 4s
1 actionable task: 1 up-to-date
```
When I run tests locally they seem to pass but Travis Ci failing on GitHub and I was wondering if you could help me understand? I am relatively new to open source and open source testing.

**Comment by Swati4star** [comment]
@RyanMarzec There are multiple unused imports:
```
[ant:checkstyle] [ERROR] /home/travis/build/project-travel-mate/Travel-Mate/Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java:25:8: Unused import - android.support.v7.widget.RecyclerView. [UnusedImports]
[ant:checkstyle] [ERROR] /home/travis/build/project-travel-mate/Travel-Mate/Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java:31:8: Unused import - android.widget.FrameLayout. [UnusedImports]
[ant:checkstyle] [ERROR] /home/travis/build/project-travel-mate/Travel-Mate/Android/app/src/main/java/io/github/project_travel_mate/SplashActivity.java:9:8: Unused import - android.view.View. [UnusedImports]
[ant:checkstyle] [ERROR] /home/travis/build/project-travel-mate/Travel-Mate/Android/app/src/main/java/io/github/project_travel_mate/mytrips/TripImageActivity.java:8:8: Unused import - android.view.View. [UnusedImports]
[ant:checkstyle] [ERROR] /home/travis/build/project-travel-mate/Travel-Mate/Android/app/src/main/java/io/github/project_travel_mate/travel/HotelsActivity.java:12:8: Unused import - android.support.annotation.Nullable. [UnusedImports]
[ant:checkstyle] [ERROR] /home/travis/build/project-travel-mate/Travel-Mate/Android/app/src/main/java/io/github/project_travel_mate/utilities/ChecklistActivity.java:11:8: Unused import - android.view.View. [UnusedImports]
[ant:checkstyle] [ERROR] /home/travis/build/project-travel-mate/Travel-Mate/Android/app/src/main/java/io/github/project_travel_mate/utilities/ChecklistActivity.java:12:8: Unused import - android.widget.FrameLayout. [UnusedImports]
[ant:checkstyle] [ERROR] /home/travis/build/project-travel-mate/Travel-Mate/Android/app/src/main/java/io/github/project_travel_mate/utilities/ChecklistActivity.java:17:8: Unused import - butterknife.ButterKnife. [UnusedImports]
```

File: Android/app/src/main/java/io/github/project_travel_mate/SplashActivity.java [hunk 15]
```diff
@@ -46,4 +53,16 @@ protected void onCreate(Bundle savedInstanceState) {
             finish();
         }, 2000);
     }
+
+    /**
+     * Viewholder for the SplashActivity item
+     */
+    class ViewHolder {
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/SplashActivity.java]: you removed viewholder from ChecklistActivity.java activity, but it's still here.
Can we have consistency in this refactoring?

**Comment by Swati4star** [comment]
Closing due to inactivity. Feel free to reopen if you continue to work on this.

**Pull request closed without merging**